lib/farscape/representor.rb

```diff
@@ -2,36 +2,93 @@
 require 'farscape/transition'
 
 module Farscape
-  class Representor
+  class SafeRepresentorAgent
     attr_reader :agent
     attr_reader :representor
 
     # TODO: Work with Representor to make this straight forwqrd.
+    # The conditional is to allow direct creation of a RepresentorAgent from a Representor
     def initialize(requested_media_type, response_body, agent)
       @agent = agent
-      if requested_media_type
-        @representor = Representors::DeserializerFactory.build(requested_media_type, response_body).to_representor
-      else
-        @representor = response_body
-      end
+      @representor = deserialize(requested_media_type, response_body)
     end
 
     def attributes
       representor.properties
     end
 
+    #TODO: Handling list of transitions
     def transitions
-      Hash[representor.transitions.map{ |trans| [trans.rel, Farscape::Transition.new(trans, agent)]}]
+      Hash[representor.transitions.map{ |trans| [trans.rel, Farscape::TransitionAgent.new(trans, agent)] }]
     end
-    
+
     def embedded
-      Hash[representor.embedded.map do |k, reps| 
-           [k, reps.map { |rep| Representor.new(false, rep, agent) }]
-      end]
+      Hash[representor.embedded.map{ |k, reps| [k, reps.map { |rep| @agent.representor.new(false, rep, agent) }] }]
     end
 
     def to_hash
       @representor.to_hash
     end
+
+    def safe
+      reframe_representor(safe=true)
+    end
+
+    def unsafe
+      reframe_representor(safe=false)
+    end
+
+    private
+
+    def reframe_representor(safe)
+      agent = safe ? @agent.safe : @agent.unsafe
+      agent.representor.new(nil, @representor, agent)
+    end
+
+    def deserialize(requested_media_type, response_body)
+      return response_body unless requested_media_type
+      Representors::DeserializerFactory.build(requested_media_type, response_body).to_representor
+    end
+
+  end
+
+  class RepresentorAgent < SafeRepresentorAgent
+    def method_missing(method, *args, &block)
+      begin
```

Comment:
You can remove `begin` and the inner `end` from this method, since rescue blocks have method scope by default.